mmdet/models/plugins/non_local.py

```diff
@@ -0,0 +1,83 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import constant_init
+
+from ..utils import ConvModule
+
+
+class NonLocalBlock2D(nn.Module):
+
+    def __init__(self,
+                 in_channels,
+                 reduction=2,
+                 conv_cfg=None,
+                 norm_cfg=None,
+                 activation=None,
+                 mode='embedded_gaussian'):
+        super(NonLocalBlock2D, self).__init__()
+        self.in_channels = in_channels
+        self.reduction = reduction
+        self.inter_channels = in_channels // reduction
+        self.mode = mode
+        assert mode in ['embedded_gaussian']
+
+        self.g = ConvModule(
```

Comment:
Adding norm layers in `g`, `theta` and `phi` seems infrequent. This needs further verification.